### PR TITLE
BBE: Add checkbox for page builds with filler content

### PR DESCRIPTION
**Pull request by aneeshd16**
Fixes Automattic/martech#1538

## Proposed Changes

Adds a checkbox to the page section of the BBE website content form to allow users to choose if the page can be built with filler (AI) content.
image

## Testing Instructions

* Apply D102972-code
* Go to `/start/do-it-for-me` and complete the purchase.
* On the website content form, open each page section and confirm that the checkbox with the label `Build this page with AI-generated text.` is present. **Note**: This checkbox will not be present on the "Contact" page.
* Hovering over the info icon next to the label should show help text in a tooltip.
* Confirm that clicking on the checkbox disables the text box above it, but it should not erase the text.
* If the section is minimized, the summary should show `AI CONTENT` instead of the entered text if the checkbox for that section is checked.
image

* Click on "Save Changes" and reload the page. Confirm that the checkboxes retain their checked state.
* Confirm that you can submit the form correctly.
* Additional test instructions in D102972-code.

## Pre-merge Checklist

- [x] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [x] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [x] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

File: client/signup/steps/website-content/wordpress-media-upload.tsx [hunk 1]
```diff
@@ -173,7 +173,7 @@ export function WordpressMediaUpload( {
 		const fileParts = fileName.split( '.' );
 		const extension = fileParts[ fileParts.length - 1 ];
 
-		if ( allowedFileTypes.includes( extension ) ) {
+		if ( allowedFileTypes.includes( extension?.toLowerCase() ) ) {
```
Comment (aneeshd16) [inline client/signup/steps/website-content/wordpress-media-upload.tsx]: A small change included in this PR. This fixes Automattic/martech#1525.

**Comment by southp** [comment]
Bumped into this warning when I reached the submitting step upon testing the backend diff:
截圖 2023-03-02 下午2 50 45

I know this PR is WIP; just wanted to share in advance in case it helps ;)

**Comment by aneeshd16** [comment]
> I know this PR is WIP; just wanted to share in advance in case it helps ;)

Thank you @southp, I have added a fix in https://github.com/Automattic/wp-calypso/pull/73859/commits/33d43b992036fed28c6867ae63b72681dfa0944f. (The error was unrelated to the changes in this PR)

File: client/signup/steps/website-content/section-types/default-page-details.tsx [hunk 2]
```diff
@@ -134,7 +150,21 @@ export function DefaultPageDetails( {
 				value={ page.content }
 				error={ formErrors[ fieldName ] }
 				label={ description }
+				disabled={ !! page.useFillerContent }
+				hasFillerContentCheckbox={ isEnglishLocale }
```
Comment (southp) [inline client/signup/steps/website-content/section-types/default-page-details.tsx:154]: I'm wondering if this is the culprit of the following warning:
image

**Review by southp — COMMENTED** [review]
It works as the test plan described :) I saw one React warning that should be fixed, though. Once that is addressed, please feel free to ping me directly for a more timely 2nd review :)

File: client/signup/steps/website-content/section-types/default-page-details.tsx (hunk 2 above)
Comment (aneeshd16) [reply client/signup/steps/website-content/section-types/default-page-details.tsx:154]: Good catch! The new prop was being forwarded to both the fieldset and textarea elements. I fixed it via https://github.com/Automattic/wp-calypso/pull/73859/commits/9533a351afada1b89d8ad53547de68633df6efff.

**Review by southp — APPROVED** [review]
I've retested it and confirmed the issue was gone. LGTM 🚢

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/7865021

Thank you @aneeshd16 for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/7865021).

**Pull request merged**